torch_xla/csrc/helpers.h

```diff
@@ -168,6 +168,24 @@ class XlaHelpers {
                                   xla::int64 expected_rank,
                                   xla::int64 offset = 0);
 
+  // Gathers the input using the order specified by the permutation. For each i,
+  // output[i] = input[permutation[i]]. The given permutation must be the same
+  // size as the input.
+  template <typename Container>
+  static std::vector<typename Container::value_type> Permute(
+      tensorflow::gtl::ArraySlice<const xla::int64> permutation,
+      const Container& input) {
+    using T = typename Container::value_type;
+    tensorflow::gtl::ArraySlice<const T> data(input);
```

Comment:
Should not be needing this step. Just use 'input'.

Reply:
This is taken from the `xla::Permute`. Don't we need it if the `input` container doesn't support random access? I don't mind dropping it, just not sure whether we want the additional generality or not.

Reply:
If it is "Slice Convertible", then it is random access ;)

Reply:
Maybe it's for the `size()` then? Anyway, done.